js/index.js

```diff
@@ -38,5 +38,57 @@ const siteContent = {
 };
 
 // Example: Update the img src for the logo
-let logo = document.getElementById("logo-img");
-logo.setAttribute('src', siteContent["nav"]["img-src"])
+// let logo = document.getElementById("logo-img");
+// logo.setAttribute('src', siteContent["nav"]["img-src"]);
+
+//nav
+
+const nav = document.querySelector('nav');
+console.log(nav.querySelectorAll('a'));
+
+nav.querySelectorAll('a')[0].textContent = siteContent.nav["nav-item-1"];
+nav.querySelectorAll('a')[1].textContent = siteContent.nav["nav-item-2"];
+nav.querySelectorAll('a')[2].textContent = siteContent.nav["nav-item-3"];
+nav.querySelectorAll('a')[3].textContent = siteContent.nav["nav-item-4"];
+nav.querySelectorAll('a')[4].textContent = siteContent.nav["nav-item-5"];
+nav.querySelectorAll('a')[5].textContent = siteContent.nav["nav-item-6"];
```

Comment:
Oof. Multiple `querySelector` calls and repeated code! Imagine if we had, say... a _hundred_ anchor tags that we needed to fill in? That'd be a lot of assignments! So how could we make that better? How could we make it more DRY?

Well, think about how you're referring to each node element and referencing it to the `nav-item` in `siteContent`?

For each index (0, 1, 2), you're referring to the nav items incremented by one (1, 2, 3). So in this instance, we could probably use a loop to assign each anchor's text content to the `index + 1` value in `siteContent`.

How would this look in code? Let's see...
```js
// select all anchor elements and convert the HTMLCollection to an array using `Array.from()`
Array.from(nav.getElementsByTagName('a')).forEach((anchor, idx) => {
  // loop over each element and use the index (idx) value to calculate the correct siteContent value
  anchor.textContent = siteContent['nav'][`nav-item-${idx+1}`];
});
```

Reply:
One other thing! While using `querySelectorAll` to grab a list of elements certainly does work, it's usually much more performant to, whenever possible, grab items using specific attribute selector methods. In this case, I'm grabbing the data using `getElementsByTagName`.

This will grab all the anchor elements inside the nav, and it's more than twice as performant than using a `querySelector` method.